contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,122 @@
+            github_username   (The username of the github bot)
+            github_token    (The personal access token for the github bot)
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = original_content.startswith('@issue')
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # gets token from dev-secrets
+        # CONFIGURE TOKEN ON LOCAL MACHINE FOR NOW AT .zuliprc-local
+        config = six.moves.configparser.ConfigParser()
+        config.read([os.path.expanduser(self.CONFIG_FILE)])
+
+        REPO_NAME = config.get('github', 'github_repo')
+        REPO_OWNER = config.get('github', 'github_repo_owner')
+
+        new_content = original_content.replace('@issue', 'by {}:'.format(original_sender,))
+        # gets the repo url
+        url_new = self.URL.format(REPO_OWNER, REPO_NAME)
+
+        # signs into github using the provided username and password
+        session = github.auth()
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        issue_content = ''
+        new_issue_title = ''
+        for part_of_title in issue_title.split():
+            if len(new_issue_title) < self.CHARACTER_LIMIT:
+                new_issue_title += '{} '.format(part_of_title)
+            else:
+                issue_content += '{} '.format(part_of_title)
+
+        new_issue_title = new_issue_title.strip()
+        issue_content = issue_content.strip()
+        new_issue_title += '...'
+
+        # Creates the issue json, that is transmitted to the github api servers
+        issue = {
+                 'title': new_issue_title,
+                 'body': '{} **Sent by [{}](https://chat.zulip.org/#) from zulip**'.format(issue_content, original_sender),
+                 'assignee': '',
+                 'milestone': 'none',
+                 'labels': [''],
+                }
+        # Sends the HTTP post request
+        r = session.post(url_new, json.dumps(issue))
+
+        if r.status_code == 201:
+            # sends the message onto the 'issues' stream so it can be seen by zulip users
+            client.send_message(dict(
+                type='stream',
+                to='issues',
+                subject=message['sender_email'],
+                # Adds a check mark so that the user can verify if it has been sent
+                content='{} :heavy_check_mark:'.format(new_content),
+            ))
```

Comment:
May be you should just `return` from here, then you will not need to handle the error case in an `else`.